test/utils.sh

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/bash
+
+WANDA_URL=${WANDA_URL:-http://localhost:4000}
+FACTS_SERVICE_URL=${FACTS_SERVICE_URL:-amqp://wanda:wanda@localhost:5672}
+DEBUG=${DEBUG:-false}
+
+function log_file {
+	echo "/tmp/checks.$(date +%Y%m%d).log"
+}
+
+function debug {
+	if [ "$DEBUG" == "true" ]; then
+		for e in "$@"; do
+			echo "[$(date '+%Y-%m-%d %H:%M:%S')] DEBUG: $e" >>"$(log_file)"
+		done
+	fi
+}
+
+function start_agent() {
+	agent_id=${1:?argument required}
+	dockerfile_path=${2:?argument required}
+	fixture_path=${3:?argument required}
+	image_name="trento_bbq_agent:$(echo -n "$fixture_path" | sha256sum | awk '{print $1}')"
+	container_name="trento_bbq_agent_$(uuidgen)"
+
+	(
+
+		debug "building image $image_name from $fixture_path"
+
+		docker build -f "$dockerfile_path" -t "$image_name" "$fixture_path"
+
+		debug "running container $container_name for agent $agent_id"
+
+		docker run -d --rm --name "$container_name" \
+			--env FACTS_SERVICE_URL="$FACTS_SERVICE_URL" \
+			--env AGENT_ID="$agent_id" \
+			--network host \
+			"localhost/$image_name"
+
+	) >/dev/null 2>&1
+
+	echo "$container_name"
+}
+
+function stop_agent() {
+	container_name=${1:?argument required}
+	docker kill "$container_name"
+}
+
+function start_check_execution() {
+	agent_id=${1:?argument required}
+	check_ids=\"${2:?argument required}\"
+	shift 2
+	for check_id in "$@"; do
+		check_ids="$check_ids,\"$check_id\""
+	done
+
+	execution_id=$(uuidgen)
+	data=$(
+		cat <<EOF
+{
+    "target_type": "cluster",
+    "env": {
+        "provider": "azure"
+    },
+    "execution_id": "$execution_id",
+    "group_id": "3dff9d03-4adf-453e-9513-8533e221bb12",
+    "targets": [
+        {
+            "agent_id": "$agent_id",
+            "checks": [$check_ids]
+        }
+    ]
+}
```

Comment:
We might need more flexibility with regards to `target_type` and `env`, `group_id` as well perhaps.
Possibly on a per-test basis.

Reply:
Definitely yes, we will need to accept all kinds of check requests. I focused on the overall test and I left this abstraction for future refinement. I didn't want to stick to a biased abstraction, for now, it just works.

Reply:
I moved the payload to the test definition, so that it's more explicit. 

https://github.com/trento-project/checks/pull/35/commits/d2a44474c47c2f86ab57e968591b3b9de24fce4b